master-firmware/src/strategy.cpp

```diff
@@ -68,13 +68,43 @@ static void strategy_forward(void* ctx, int relative_distance_mm)
 static bool strategy_goto_xya(void* ctx, int x_mm, int y_mm, int a_deg)
 {
     strategy_context_t* strat = (strategy_context_t*)ctx;
-    return strategy_goto_avoid(strat, x_mm, y_mm, a_deg, TRAJ_FLAGS_ALL);
+
+    motion_planner_set_goal(x_mm, y_mm, a_deg);
+    auto end_reason_xy = trajectory_wait_for_end(TRAJ_FLAGS_ALL);
+
+    if (end_reason_xy == TRAJ_END_GOAL_REACHED) {
+        // wait again for rotation
+        strat->wait_ms(200);
+        auto end_reason_a = trajectory_wait_for_end(TRAJ_FLAGS_ALL);
```

Comment:
Why do you wait again when the end_reason doesn't change?

Reply:
it's mentioned in the comments, we need to wait for the final rotation, still don't like this, it's flaky IMO

Reply:
Do I understand right that `end_reason_a` is `TRAJ_END_GOAL_REACHED` independent of the angle? (depends only on xy)

Reply:
`TRAJ_END_GOAL_REACHED` is returned when either of trajectory setpoints is reached (on distance and angle) this covers the last xy setpoint before reaching a goal and the last rotation we do after reaching the xy setpoint.
see https://github.com/cvra/robot-software/blob/master/master-firmware/src/robot_helpers/trajectory_helpers.c#L42-L44